### PR TITLE
Conda tweaks and Fix Example Notebook 1

### PR DESCRIPTION
**Pull request by aufdenkampe**
This PR:
- Updates `environment.yml` Python 3.10 and to pip install `hsp2` directly from PyPI after creating the conda environment with all dependencies for running hsp2 and the example notebooks, all from one command:
  ```console
  conda env create --file=environment.yml --solver=libmamba
  ```
- Updates ReadMe to include instructions for above
- Updates the `examples/1_Intro_to_HSP2.ipynb` Notebook to work with the repo 'src' directory reorganization for v0.11.0a1, the new environment file above, and on anyone's computer.
- Temporarily removes `__version__` from `src/hsp2/hsp2/__init__.py`, which will require a future fix to return the ability of the Python kernel to retrieve the version number. Note that the version can still be retrieved from conda.
  - This was throwing an error when running in development mode with `conda develop`.

Accepting this PR would also close:
- #100

### CODE REVIEW

**Comment by aufdenkampe** [comment]
@PaulDudaRESPEC, I think we should merge this PR into `develop` as soon as possible, so that others can run the example with IOManager as a followup on the clarifications in https://github.com/respec/HSPsquared/issues/182#issuecomment-2542252660

**Comment by timcera** [comment]
I did change the hard-coded version string to something dynamic, however, I don't think is has a use and could be removed (testing to make sure of course).  I kept it there not because I thought it was useful, but because I didn't want to be disruptive.  I think the idea of the __version__ string is inherited from ancient Python project "skeletons" so that you can make bug reports (maybe that is the use case?), but hsp2 is a low-volume development community and I think a bit of overkill to have the version string programmatically available.  As you pointed out you can easily get the installed version through pip or conda. 

Basically, if the version string needs to come out to make things smoother for "conda develop" then take it out.

**Comment by aufdenkampe** [comment]
@PaulDudaRESPEC, thanks for merging this. I just found a fix for the versions thing and cued up this PR for you to review:
- #185
